File: projects/3/jaylon_williams/extract_tables.jl
# Import necessary modules
using Downloads
using DelimitedFiles

# Step 1: Download the HTML file
url = "https://en.wikipedia.org/wiki/List_of_largest_companies_in_the_United_States_by_revenue"
html_file = "page.html"
Downloads.download(url, html_file)
println("Downloaded HTML file.")

# Step 2: Read HTML content
html_content = read(html_file, String)

# Step 3: Extract table data (very basic string parsing approach)
tables = []
table_start = "<table"
table_end = "</table>"

start_idx = findnext(table_start, html_content, 1)
while start_idx !== nothing
    end_idx = findnext(table_end, html_content, start_idx + length(table_start) - 1)
    if end_idx === nothing
        break
    end
    push!(tables, html_content[start_idx:end_idx + length(table_end) - 1])
    start_idx = findnext(table_start, html_content, end_idx + length(table_end) - 1)
end

# Step 4: Save extracted tables into CSV files
for (i, table) in enumerate(tables)
    csv_file = "table_$(i).csv"
    open(csv_file, "w") do f
        # Basic conversion: Remove HTML tags and replace with commas
        table_clean = replace(table, r"<[^>]*>" => ",")
        table_clean = replace(table_clean, r",+" => ",")  # Remove extra commas
        write(f, table_clean)  # Use write instead of println
    end
    println("Saved table to: ", csv_file)
end

println("Extraction complete! CSV files created.")

File: projects/3/skylar_sawyer/extract_tables.R
# Define URL of the Wikipedia page
url <- "https://en.wikipedia.org/wiki/List_of_largest_companies_in_the_United_States_by_revenue"

# Download the HTML file
html_file <- "page.html"
download.file(url, html_file)

# Read the HTML file
html_content <- readLines(html_file, warn = FALSE)

# Extract tables using regular expressions (since we're not using external packages)
#tables <- regmatches(html_content, gregexpr("<table.*?</table>", html_content, perl = TRUE))[[1]]

tables <- html_content[grep("<table", html_content)]
if (length(tables) == 0) {
  stop("No tables found in the HTML file.")
} else {
  print(paste("Extracted", length(tables), "tables."))
}

# Save each table to a CSV file
for (i in seq_along(tables)) {
  table_file <- paste0("table_", i, ".csv")
  cat("Saved:", table_file, "\n")
  writeLines(tables[i], table_file)
}

if (file.exists(html_file)) {
  cat("Downloaded HTML file successfully.\n")
} else {
  cat("Failed to download HTML file.\n")
}
